include/treelite/tree_impl.h

```diff
@@ -589,16 +614,12 @@ Tree<ThresholdType, LeafOutputType>::SerializeToFile(FILE* dest_fp) {
 
 template <typename ThresholdType, typename LeafOutputType>
 inline void
-Tree<ThresholdType, LeafOutputType>::InitFromPyBuffer(std::vector<PyBufferFrame>::iterator begin,
-                                                      std::vector<PyBufferFrame>::iterator end) {
-  if (std::distance(begin, end) != kNumFramePerTree) {
-    throw Error("Wrong number of frames specified");
-  }
-  auto scalar_handler = [&begin](auto* field) {
-    InitScalarFromPyBuffer(field, *begin++);
+Tree<ThresholdType, LeafOutputType>::InitFromPyBuffer(std::vector<PyBufferFrame>::iterator& it) {
+  auto scalar_handler = [&it](auto* field) {
+    InitScalarFromPyBuffer(field, *it++);
   };
-  auto array_handler = [&begin](auto* field) {
-    InitArrayFromPyBuffer(field, *begin++);
+  auto array_handler = [&it](auto* field) {
+    InitArrayFromPyBuffer(field, *it++);
```

Comment:
```suggestion
    InitArrayFromPyBuffer(field, *(it++));
```
As above. I would recommend this wherever it appears.